### PR TITLE
#573: Display prices even before login

### LINKED ISSUE #573: Display price in guest mode (Demo feedback)

### PR DESCRIPTION
**Pull request by hani-ibrahim**
fixes #573

### CODE REVIEW

**Comment by hani-ibrahim** [comment]
👍

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift [hunk 1]
```diff
@@ -66,7 +66,7 @@ extension CheckoutSummaryDataModel {
         self.billingAddress = addresses?.billingAddress ?? cartCheckout?.checkout?.billingAddress
         self.paymentMethod = cartCheckout?.checkout?.payment.selected?.method
         self.shippingPrice = 0
-        self.totalPrice = cartCheckout?.cart?.grossTotal.amount
+        self.totalPrice = cartCheckout?.cart?.grossTotal.amount ?? selectedArticleUnit.unit.price.amount
```
Comment (bonkey) [inline AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift]: Logical would be to add `shippingPrice` here.

File: AtlasUI/AtlasUITests/LoggedInSummaryActionHandlerTests.swift [hunk 2]
```diff
@@ -182,7 +182,7 @@ extension LoggedInSummaryActionHandlerTests {
                         let selectedArticleUnit = SelectedArticleUnit(article: article, selectedUnitIndex: 0)
                         LoggedInSummaryActionHandler.createInstance(customer, selectedUnit: selectedArticleUnit) { result in
                             guard let actionHandler = result.process() else { return fail() }
-                            let dataModel = CheckoutSummaryDataModel(selectedArticleUnit: selectedArticleUnit)
+                            let dataModel = CheckoutSummaryDataModel(selectedArticleUnit: selectedArticleUnit, totalPrice: selectedArticleUnit.unit.price.amount)
```
Comment (bonkey) [inline AtlasUI/AtlasUITests/LoggedInSummaryActionHandlerTests.swift]: Used at least twice (one in `SizeListSelectionViewController.swift`), so maybe there should be a shortcut `selectedArticleUnit.priceValue` or sth?

**Comment by bonkey** [comment]
👍

File: AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift (hunk 1 above)
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/CheckoutSummaryDataModel.swift]: if we don't have a cart then we will not have delivery method when means no shipping price

**Pull request merged**